### PR TITLE
Fix cuda/cpu check on NoneType

### PR DESCRIPTION
**Pull request by mikekgfb**
Summary: Fix cuda/cpu check on NoneType

Test Plan: sabdcastle/ github CI/CD

Differential Revision: D41203955

### CODE REVIEW

**Review by drisspg — APPROVED** [review]
👍

File: torch/nn/modules/activation.py [hunk 1]
```diff
@@ -1114,7 +1114,7 @@ def forward(self, query: Tensor, key: Tensor, value: Tensor, key_padding_mask: O
             # generator expressions.
             if torch.overrides.has_torch_function(tensor_args):
                 why_not_fast_path = "some Tensor argument has_torch_function"
-            elif not all([(x.is_cuda or 'cpu' in str(x.device)) for x in tensor_args]):
+            elif not all([(x is None or x.is_cuda or 'cpu' in str(x.device)) for x in tensor_args]):
```
Comment (ngimel) [inline torch/nn/modules/activation.py:1117]: pre-existing condition, but proper check is x.device.type=='cpu'

**Review by ngimel — APPROVED** [review]

**Comment by cpuhrsch** [comment]
Also addresses https://github.com/pytorch/pytorch/issues/88669

**Pull request closed without merging**